### PR TITLE
Web 1426  add from wishlist

### PR DESCRIPTION
**Pull request by jnurse**
This PR adds the ability to add an item to your cart from your wishlist. 

 **JIRA**: https://mobify.atlassian.net/browse/WEB-1426
 **Linked PRs**: https://github.com/mobify/progressive-web-sdk/pull/779

## Changes
- Added UI for add to cart success
- Added Ui for add to cart failure
- refactored contents of product details item added modal into a re-usable component
- Added merlins command for addToCartFromWishlist
- Added SFCC commend to addToCartFromWishlist & removeItemFromWishlist 

## To dos
- [ ] (if applicable) analytics events instrumented to measure impact of change
- [ ] Add a high-level description of your changes to the CHANGELOG.md, including a link to the PR with your changes
- [x] Add command to stub connector

## How to test-drive this PR
- checkout the add-from-wishlist-command branch of the SDK
- run `npm run dev:build`
- run `npm link`
- cd into the scaffold web folder
- run `npm link progressive-web-sdk`
- preview the merlin's site
- sign in to an account with items in the wishlist
- view the wishlist page
- add an item from the wishlist to the cart
- the item should be added to the cart and removed from the wishlist
- repeat using the SFCC connector & previewing the SFCC site

### CODE REVIEW

File: web/app/modals/wishlist-item-added/container.jsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    thumbnail: getProductThumbnail,
+    open: isModalOpen(WISHLIST_ITEM_ADDED_MODAL),
+    quantity: getWishlistItemQuantity,
+    title: getProductTitle,
+    price: getProductPrice
+})
+
+const mapDispatchToProps = {
+    onGoToCheckout: wishlistActions.goToCheckout,
+    onDismiss: stripEvent(() => closeModal(WISHLIST_ITEM_ADDED_MODAL, UI_NAME.addToCart))
```
Comment (wizardlyhel) [inline web/app/modals/wishlist-item-added/container.jsx]: This one should be `UI_NAME.wishlist`

Slightly confused ... What's this one for? https://github.com/mobify/platform-scaffold/pull/908/files#diff-366ba221739fc843ee6de373dac13f3fR35 (The `goToCheckout` function)

File: web/app/containers/wishlist/partials/wishlist-items.jsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                                href={'test/url'}
+                                data-analytics-name={UI_NAME.editItem}
+                                onClick={() => console.log('test edit button')}
+                                >
+                                Edit
+                            </Button>
+                            <Button
+                                className="u-text-size-small u-color-brand u-text-letter-spacing-normal qa-cart__remove-item"
+                                innerClassName="u-padding-end-0 u-padding-bottom-0 u-padding-start-0"
+                                onClick={() => console.log('test remove button')}
```
Comment (jnurse) [inline web/app/containers/wishlist/partials/wishlist-items.jsx]: The remove button hasn't been implemented yet. The console log is just a placeholder until we work on the ticket for removing wishlist items.

File: web/app/store/selectors.js [hunk 3]
```diff
@@ -3,7 +3,6 @@
 /* * *  *  * *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  * */
 
 export const getUi = ({ui}) => ui
-export const getUser = ({user}) => user
```
Comment (Cdok) [inline web/app/store/selectors.js]: Looks like we're missing 1 updated import path
```
WARNING in ./app/containers/account-info/selectors.js
    8:56-63 "export 'getUser' was not found in '../../store/selectors'
```

**Comment by jnurse** [comment]
Good catch on the duplicate products/wrong ID thing. Should be fixed now.

**Comment by Cdok** [comment]
screen shot 2017-08-01 at 10 49 26 am

If I move the quantity stepper up to `3` on the Wishlist, then add to cart, only 1 of the item gets added to my cart

**Comment by jnurse** [comment]
@Cdok Updating an item's quantity in the wishlist is part of the edit wishlist ticket, so I haven't hooked up the item steppers in this PR. Once it's set up and updates the item in the redux store then this command should use the correct quantity.

**Pull request merged**